modules/explorer-api/src/main/scala/org/ergoplatform/explorer/http/api/v1/services/Boxes.scala

```diff
@@ -144,6 +156,36 @@ object Boxes {
         .thrushK(trans.xa)
     }
 
+    /** Get all outputs with the given `address` in proposition & filter outputs spent in the mempool (for unconfirmed transactions)
+      */
+    private def getUnspentOutputsByAddressF(
+      address: Address,
+      ord: SortOrder,
+      excludedBoxes: Option[NonEmptyList[BoxId]]
+    ): F[List[OutputInfo]] = {
+      val ergoTree = addressToErgoTreeHex(address)
+      (
+        for {
+          unspentOuts <- outputs
+                           .streamUnspentByErgoTree(ergoTree, ord.value, excludedBoxes)
+                           .chunkN(serviceSettings.chunkSize)
+                           .through(toOutputInfo)
+                           .to[List]
+        } yield unspentOuts
+      ) ||> trans.xa
+    }
+
+    def `getUnspent&UnconfirmedOutputsMergedByAddress`(
+      address: Address,
+      ord: SortOrder,
+      getUSpentBoxes: Address => F[List[BoxId]],
+      getUOutputs: Address => F[List[UOutputInfo]]
```

Comment:
It seems those methods should be implemented in the same service rather than in a separate one and passed as functions.